src/main/scala/com/prezi/haskell/gradle/tasks/StoreDependentSandboxes.scala

```diff
@@ -23,26 +29,76 @@ class StoreDependentSandboxes extends DefaultTask with HaskellDependencies {
     needsConfigurationSet
 
     getLogger.info("Storing dependent sandboxes for {}", getProject.getName)
-    for (dependency <- configuration.get.getResolvedConfiguration.getFirstLevelModuleDependencies.asScala) {
-      storeDependency(dependency)
-    }
+    val storeDependencyResults =
+      for (dependency <- configuration.get.getResolvedConfiguration.getFirstLevelModuleDependencies.asScala) yield {
+        storeDependency(dependency)
+      }
+
+    _isAnySandboxUpdated = storeDependencyResults.exists { _._2 == SandBoxStoreResult.Created }
   }
 
-  def storeDependency(dependency: ResolvedDependency, prefix: String = ""): Set[SandboxArtifact] = {
+  def storeDependency(dependency: ResolvedDependency, prefix: String = ""): (Set[SandboxArtifact], SandBoxStoreResult) = {
+
     getLogger.info("{}Dependency {}", prefix, dependency.getName)
-    val depSandboxes =
-      (for (child <- dependency.getChildren.asScala;
-            childDep <- storeDependency(child, prefix + "  "))
-       yield childDep).toSet
+
+    val (depSandboxes: Set[SandboxArtifact], aggregatedStoreResult: SandBoxStoreResult) =
+      foldStoreDependencyResults(
+        (for {
+          child <- dependency.getChildren.asScala
+          storeDependencyResult <- List(storeDependency(child, prefix + "  "))
+        } yield storeDependencyResult).toSet
+      )
 
     val sandboxes = dependency.getAllModuleArtifacts
       .asScala
       .map(artifact => new SandboxArtifact(artifact.getName, artifact.getFile))
       .toSet
-    for (sandbox <- sandboxes) {
-      store.store(sandbox, depSandboxes)
+
+    val sandboxStoreResults =
+      for (sandbox <- sandboxes) yield {
+        memoizedStoreDependencyInStore(sandbox, depSandboxes)
+      }
+
+    (sandboxes,
+      if (sandboxStoreResults.contains(SandBoxStoreResult.Created) || aggregatedStoreResult == SandBoxStoreResult.Created)
+        SandBoxStoreResult.Created
+      else
+        SandBoxStoreResult.AlreadyExists)
+  }
+
+  def foldStoreDependencyResults(results: Set[(Set[SandboxArtifact], SandBoxStoreResult)]): (Set[SandboxArtifact], SandBoxStoreResult) =
+    results.foldLeft((Set[SandboxArtifact](), SandBoxStoreResult.AlreadyExists.asInstanceOf[SandBoxStoreResult])) { (acc, elem) =>
+      (acc._1 ++ elem._1,
```

Comment:
I think using `._1` and `._2` here (and everywhere in this changeset) is very ugly; if you need to access the tuple's items you should create a record with named fields, if you use it as a tuple do pattern matching and give names to the items..